src/rqt_reconfigure/node_selector_widget.py

```
@@ -225,7 +225,6 @@ def _selection_selected(self, index_current, rosnode_name_selected):
         # Only when it's a terminal we move forward.
 
         item_child = self._nodeitems[rosnode_name_selected]
-        item_widget = None
```

I don't think this change is correct, or, at least, this definitely changes the behavior.

If `item_child.get_param_client_widget()` throws an exception, now `item_widget` will throw a `NameError` below, rather than being a "None" in the debug log (and being emitted).

Could you explain the reasoning behind this change?

---

Your comments are all about changes suggested by the IDE. In particular to this one, item_widget always takes whatever value is returned by `item_child.get_param_client_widget()`. Notice that the exception is re-raised; the code does not continue in case of an exception.

---

Oh, really good point.  Sorry, I missed that.

Given that we are doing a `try..except Exception as e: raise e`, I think we can just get rid of the `try..except` block completely; that is the default behavior.